raytracing/__main__.py

```diff
@@ -9,28 +9,50 @@
 from . import olympus
 from . import utils
 
+import os 
+import sys
 import argparse
+import re
+
+pythonExec = sys.executable        
+topDir = os.path.dirname(os.path.realpath(__file__))
+examplesDir = "{0}/../examples".format(topDir)
+exampleScriptNames = [f for f in os.listdir(examplesDir) if re.match(r'.*.py$', f)]
+exampleScriptNames.sort()
+exampleScripts = {name:"{0}/{1}".format(examplesDir, name) for name in exampleScriptNames}
 
 ap = argparse.ArgumentParser(prog='python -m raytracing')
 ap.add_argument("-e", "--examples", required=False, default='all',
                 help="Specific example numbers, separated by a comma")
 ap.add_argument("-c", "--classes", required=False, action='store_const',
                 const=True, help="Print the class hierarchy in graphviz format")
+ap.add_argument("-l", "--list", required=False, action='store_const',
+                const=True, help="List all the accessible examples")
 
 args = vars(ap.parse_args())
 examples = args['examples']
 printClasses = args['classes']
+listExamples = args['list']
 
 if printClasses:
     printClassHierarchy(Rays)
     printClassHierarchy(Matrix)
     exit(0)
-
+elif listExamples:
+    for i, name in enumerate(exampleScriptNames):
+        print("{0:2d}. {1}".format(i+1, name))
+    exit(0)
 if examples == 'all':
     examples = range(1, 30)
 else:
     examples = [int(y) for y in examples.split(',')]
 
+if 0 in examples:
```

Comment:
would it be `if examples:`
and removing all the other examples in the __main__ ?

Reply:
Yes I think you had an old version